resolver/dns_resolver_cache.ml

```diff
@@ -152,7 +152,7 @@ let resolve t ~rng ip_proto ts name typ =
   *)
   let rec go t types name =
     Logs.debug (fun m -> m "go %a" Domain_name.pp name) ;
-    match find_nearest_ns rng ip_proto ts t (Domain_name.raw name) with
+    match find_nearest_ns rng ip_proto dnssec ts t (Domain_name.raw name) with
     | `NeedAddress ns -> go t addresses ns
     | `NeedDnskey (zone, ip) -> zone, zone, [`K (Rr_map.K Dnskey)], ip, t
```

Comment:
`` `NeedDnskey`` should never be returned if `dnssec` is false. I don't think it's worth it to try encode it in the types, though.

Reply:
indeed.